dask/dataframe/io/parquet.py

```diff
@@ -121,7 +117,8 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
                         columns=[c for c in pf.columns if c in dtypes])
 
     for cat in categories:
```

Comment:
It is not possible to know whether the data is really stored as categorical ("dictionary encoding") before reading it. If the user specifies a column as categorical but any part of it is not, there will be an error on reading.